src/patternfly/components/FileUpload/examples/FileUpload.md

```diff
@@ -0,0 +1,147 @@
+---
+title: File upload
+section: experimental
+cssPrefix: pf-c-file-upload
+---
+
+## Examples
+```hbs title=Basic-file-upload
+{{#> file-upload file-upload--type="form"}}
+  {{#> input-group}}
+    {{#> form-control controlType="input" input="true" form-control--attribute='id="basic-file-upload" name="basic-file-upload" placeholder="Drag a file here or browse to upload" aria-label="upload a file"'}}
+    {{/form-control}}
+    {{#> button button--modifier="pf-m-control"}}
+      Browse...
+    {{/button}}
+    {{#> button button--modifier="pf-m-control" button--attribute="disabled"}}
+      Clear
+    {{/button}}
+  {{/input-group}}
+  {{#> file-upload-text-area file-upload-text-area--attribute='aria-label="text-area-empty"'}}
+  {{/file-upload-text-area}}
+{{/file-upload}}
+```
+
+```hbs title=Browse-to-upload-file
+{{#> file-upload file-upload--type="form"}}
+  {{#> input-group}}
+    {{#> form-control controlType="input" input="true" form-control--attribute='id="basic-file-upload-browse" name="basic-file-upload-browse" placeholder="Drag a file here or browse to upload" aria-label="select browse button to upload file"'}}
+    {{/form-control}}
```

Comment:
What about a wrapping element for the input group, one for the text area, and one for the message? Like `pf-c-form-control__header` or something like that, `__main/content` for the textarea container, and `__footer` for the message.

Reply:
I think wrapping them is a good idea, not too sure about the naming though. I think that the file upload component is always going to have an input group at the top and a text area underneath, could you think of a situation where this wouldn't be the case?

@mattnolting what are your thoughts here?